frontend/lib/signup/view/signup.view.dart

```diff
@@ -0,0 +1,188 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:provider/provider.dart';
+import 'package:utm_marketplace/shared/themes/theme.dart';
+import 'package:utm_marketplace/signup/view_models/signup.viewmodel.dart';
+
+class SignUp extends StatefulWidget {
+  const SignUp({super.key});
+
+  @override
+  SignUpState createState() => SignUpState();
+}
+
+class SignUpState extends State<SignUp> {
+  final _formKey = GlobalKey<FormState>();
+  bool _obscureText = true;
+
+  @override
+  Widget build(BuildContext context) {
+    final signUpViewModel = Provider.of<SignUpViewModel>(context);
+
+    final logoText = StyleText(
+      text: 'UTMarketplace',
+      style: ThemeText.header.copyWith(
+        color: Theme.of(context).colorScheme.primary,
+      ),
+    );
+
+    final emailField = TextFormField(
+      decoration: InputDecoration(
+      border: OutlineInputBorder(),
+      labelText: 'Enter your UofT email',
+      ),
+      onChanged: (value) {
+      signUpViewModel.email = value;
+      },
+      validator: (value) {
+      if (value == null || value.isEmpty) {
+        return 'Please enter a valid UofT email address';
+      }
+      final emailRegex = RegExp(r'^[a-zA-Z0-9._%+-]+@(mail\.utoronto\.ca|utoronto\.ca)$');
+      if (!emailRegex.hasMatch(value)) {
+        return 'Please enter a valid UofT email address';
+      }
+      return null;
+      },
+    );
+
+    final passwordField = TextFormField(
+      obscureText: _obscureText,
+      decoration: InputDecoration(
+        border: OutlineInputBorder(),
+        labelText: 'Enter your password',
+        suffixIcon: IconButton(
+          icon: Icon(
+            _obscureText ? Icons.visibility_off : Icons.visibility,
+          ),
+          onPressed: () {
+            setState(() {
+              _obscureText = !_obscureText;
+            });
+          },
+        ),
+      ),
+      onChanged: (value) {
+        signUpViewModel.password = value;
+      },
+      validator: (value) {
+        if (value == null || value.isEmpty) {
+          return 'Please enter a password';
+        }
+        return null;
+      },
+    );
+
+    final confirmPasswordField = TextFormField(
+      obscureText: _obscureText,
+      decoration: InputDecoration(
+        border: OutlineInputBorder(),
+        labelText: 'Confirm your password',
+      ),
+      validator: (value) {
+        if (value == null || value.isEmpty) {
+          return 'Please confirm your password';
+        }
+        if (value != signUpViewModel.password) {
+          return 'Passwords do not match';
+        }
+        return null;
+      },
+    );
+
+    final signupButton = ElevatedButton(
```

Comment:
This is better done with a `_buildSignUpButton` helper b/c you need to pass in the `context` object that you'll receive from the `build` function. 

In Flutter, the `context` tells you where in the Widget tree you are and the SnackBar needs to know this. Your code here is a bit finicky b/c you're referencing `context` in a scope where context isn't defined. TL;DR `context` is not a global value, you need to take it from the Widget's `build` function.